anaUltraScurve.py

```diff
@@ -0,0 +1,356 @@
+            if noise[0] > 20.0 or ped_eff[0] > 50.0: mask[0] = True
+            else: mask[0] = False
+            masks[vfat].append(mask[0])
+            chi2[0] = scanFits[3][vfat][chan]
+            ndf[0] = int(scanFits[5][vfat][chan])
+            holder_curve = vScurves[vfat][chan]
+            holder_curve.Copy(scurve_h)
+            Nhigh[0] = int(scanFits[4][vfat][chan])
+            #Filling the arrays for plotting later
+            if options.drawbad:
+                if (Chi2 > 1000.0 or Chi2 < 1.0):
+                    overlay_fit(vfat, chan)
+                    print "Chi2 is, %d"%(Chi2)
+                    pass
+                pass
+            myT.Fill()
+            pass
+        if not (options.channels or options.PanPin):
+            fitSums[vfat] = r.TGraphErrors(len(fitThr),np.array(stripList),np.array(fitThr),np.zeros(len(fitThr)),np.array(fitENC))
+            fitSums[vfat].SetTitle("VFAT %i Fit Summary;Strip;Threshold [fC]"%vfat)
+            pass
+        elif options.channels:
+            fitSums[vfat] = r.TGraphErrors(len(fitThr),np.array(chanList),np.array(fitThr),np.zeros(len(fitThr)),np.array(fitENC))
+            fitSums[vfat].SetTitle("VFAT %i Fit Summary;Channel;Threshold [fC]"%vfat)
+            pass
+        elif options.PanPin:
+            fitSums[vfat] = r.TGraphErrors(len(fitThr),np.array(panList),np.array(fitThr),np.zeros(len(fitThr)),np.array(fitENC))
+            fitSums[vfat].SetTitle("VFAT %i Fit Summary;Panasonic Pin;Threshold [fC]"%vfat)
+            pass
+        
+        fitSums[vfat].SetName("fitSum%i"%vfat)
+        fitSums[vfat].SetMarkerStyle(2)
+        pass
+    pass
+
+canv = r.TCanvas('canv','canv',500*8,500*3)
+legend = r.TLegend(0.75,0.7,0.88,0.88)
+if not options.PanPin:
+    canv.Divide(8,3)
+    r.gStyle.SetOptStat(0)
+    for vfat in range(0,24):
+        r.gStyle.SetOptStat(0)
+        canv.cd(vfat+1)
+        vSum[vfat].Draw('colz')
+        if options.IsTrimmed:
+            legend.Clear()
+            legend.AddEntry(line, 'trimVCal is %f'%(trimVcal[vfat]))
+            legend.Draw('SAME')
+            print trimVcal[vfat]
+            lines[vfat].SetLineColor(1)
+            lines[vfat].SetLineWidth(3)
+            lines[vfat].Draw('SAME')
+            pass
+        canv.Update()
+        pass
+    pass
+else:
+    canv.Divide(8,6)
+    r.gStyle.SetOptStat(0)
+    for ieta in range(0,8):
+        for iphi in range (0,3):
+            r.gStyle.SetOptStat(0)
+            canv.cd((ieta+1 + iphi*16)%48 + 16)
+            vSum[ieta+(8*iphi)].Draw('colz')
+            canv.Update()
+            canv.cd((ieta+9 + iphi*16)%48 + 16)
+            vSum2[ieta+(8*iphi)].Draw('colz')
+            canv.Update()
+            pass
+        pass
+    pass
+
+canv.SaveAs(filename+'/Summary.png')
+
+canv = r.TCanvas('canv','canv',500*8,500*3)
+canv.Divide(8,3)
+r.gStyle.SetOptStat(0)
+for vfat in fitSums.keys():
+    r.gStyle.SetOptStat(0)
+    canv.cd(vfat+1)
+    fitSums[vfat].Draw('ap')
```

Comment:
It is an easy fix anyways right? Just put the if protection like the rest of the blocks for the fitting option. The idea is without -f it only makes the Summary plot and runs faster since it is skipping all the fitting part. This plot doesn't make sense unless you do the fits, I just forgot to put the if part to protect it.